x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/SqlDataTypes.java

```diff
@@ -83,6 +87,32 @@
     public static final DataType GEO_SHAPE = new DataType("geo_shape", Integer.MAX_VALUE, false, false, false);
     public static final DataType GEO_POINT = new DataType("geo_point", Double.BYTES * 2,  false, false, false);
     public static final DataType SHAPE =     new DataType("shape",     Integer.MAX_VALUE, false, false, false);
+
+    // arrays
+    public static final DataType BOOLEAN_ARRAY      = of(BOOLEAN);
+    // integer numeric
+    public static final DataType BYTE_ARRAY         = of(BYTE);
+    public static final DataType SHORT_ARRAY        = of(SHORT);
+    public static final DataType INTEGER_ARRAY      = of(INTEGER);
+    public static final DataType LONG_ARRAY         = of(LONG);
+    // decimal numeric
+    public static final DataType DOUBLE_ARRAY       = of(DOUBLE);
+    public static final DataType FLOAT_ARRAY        = of(FLOAT);
+    public static final DataType HALF_FLOAT_ARRAY   = of(HALF_FLOAT);
+    public static final DataType SCALED_FLOAT_ARRAY = of(SCALED_FLOAT);
+    // string
+    public static final DataType KEYWORD_ARRAY      = of(KEYWORD);
+    public static final DataType TEXT_ARRAY         = of(TEXT);
+    // date
+    public static final DataType DATETIME_ARRAY     = of(DATETIME);
```

Comment:
maybe add a comment that since DATE/TIME are SQL only types (not coming from index fields directly) currently arrays of those 2 types are not supported.